src/main/java/appeng/client/render/NetworkVisualiserRender.java

```diff
@@ -0,0 +1,344 @@
+/*
+ * Copyright (c) bdew, 2014 - 2015 https://github.com/bdew/ae2stuff This mod is distributed under the terms of the
+ * Minecraft Mod Public License 1.0, or MMPL. Please check the contents of the license located in
+ * http://bdew.net/minecraft-mod-public-license/
+ */
+
+package appeng.client.render;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.entity.EntityPlayerSP;
+import net.minecraft.client.gui.FontRenderer;
+import net.minecraft.client.renderer.Tessellator;
+import net.minecraft.client.renderer.entity.RenderManager;
+import net.minecraft.item.ItemStack;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.MovingObjectPosition;
+import net.minecraftforge.client.event.RenderWorldLastEvent;
+
+import org.lwjgl.opengl.GL11;
+
+import appeng.api.config.Settings;
+import appeng.api.util.DimensionalCoord;
+import appeng.items.tools.ToolNetworkVisualiser;
+import appeng.items.tools.ToolNetworkVisualiser.VLink;
+import appeng.items.tools.ToolNetworkVisualiser.VLinkFlags;
+import appeng.items.tools.ToolNetworkVisualiser.VNode;
+import appeng.items.tools.ToolNetworkVisualiser.VNodeFlags;
+import appeng.items.tools.ToolNetworkVisualiser.VisualisationModes;
+import appeng.tile.networking.TileWirelessConnector;
+import appeng.tile.networking.TileWirelessHub;
+import cpw.mods.fml.common.eventhandler.SubscribeEvent;
+
+public class NetworkVisualiserRender {
+
+    private static long expTime;
+    private final double SIZE = 0.2d;
+    private final int staticList = GL11.glGenLists(1);
+    private static boolean needListRefresh = true;
+    private static ArrayList<VNode> vNodeSet = new ArrayList<>();
+    private static ArrayList<VLink> vLinkSet = new ArrayList<>();
+    private static final Set<VLink> dense = new HashSet<>();
+    private static final Set<VLink> normal = new HashSet<>();
+    private static VisualisationModes mode = VisualisationModes.FULL;
+    private static final Minecraft mc = Minecraft.getMinecraft();
+    private static boolean renderWireless = false;
+    private static DimensionalCoord otherDc;
+    private static final List<DimensionalCoord> hubConnections = new ArrayList<>();
+    private static DimensionalCoord prevPos;
+
+    ArrayList<VisualisationModes> renderNodesModes = new ArrayList<>(
+            Arrays.asList(VisualisationModes.NODES, VisualisationModes.FULL, VisualisationModes.NONUM));
+
+    ArrayList<VisualisationModes> renderLinksModes = new ArrayList<>(
+            Arrays.asList(
```

Comment:
Are these ever edited, or have a reason to copy the initial lists? If not instead you could just do `List<VisualisationModes>` rather than `ArrayList`